berkdb/crypto/crypto.c

```diff
@@ -22,6 +22,36 @@ static const char revid[] = "$Id: crypto.c,v 1.23 2003/06/30 17:19:41 bostic Exp
 #include "dbinc/db_page.h"
 #include "dbinc/crypto.h"
 
+#ifndef HAVE_CRYPTO
+/**
+ * __crypto_compare --
+ * 	returns zero iff `len` bytes at `src` and `tgt` are equal. Time is independent
+ *	of the contents of `src` and `tgt`.
+*/
+int
+__crypto_compare(src, dst, len)
+	const void *src;
+	const void *dst;
+	size_t len;
+{
+	const unsigned char *a = (const unsigned char *)src;
+	const unsigned char *b = (const unsigned char *)dst;
```

Comment:
I don't think the explicit type conversion is needed: a void pointer can be safely promoted to any other types. and both sides have `const` qualifier.